app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt

```diff
@@ -139,8 +139,12 @@ fun ViewFriendsScreen(
   // New state variable for Friend Requests expansion
   var isFriendRequestsExpanded by remember { mutableStateOf(true) }
 
-  // Fetch pending battles when the screen is composed
-  LaunchedEffect(Unit) { battleViewModel?.fetchPendingBattlesForUser() }
+  LaunchedEffect(Unit) {
+    userProfileViewModel.userProfile.value?.uid?.let { uid ->
+      userProfileViewModel.startListeningToUserProfile(uid)
+    }
```

Comment:
Hi Mathis, 
Thanks for your suggestion.
I tried doing this: "shouldn't the listener be set only once when the app is launched ? (In MainActivity.kt ?)" but it does not work. In fact, once you enter the friends screen, it does not get updated even when you close the screen and reopen it (which was the case before this PR). I think this is because the friends screen takes the userProfileViewModel as a val. I am currently working on a solution.

Reply:
You are totally right. I decided to call the startListeningToUserProfile once the user successfully signs in. With that, the listener is only set once when he logs in.